### PR TITLE
[PyROOT][ROOT-10693] Add __iter__ pythonization for iterators

### PR DESCRIPTION
**Pull request by stwunsch**
Returning for an iterator itself with `__iter__` ensures that we can
iterate over an iterator with an iterator, which happens for example
if you call `for x in iter(foo)`.

The upstream PR can be found here: https://bitbucket.org/wlav/cpycppyy/pull-requests/33/

### CODE REVIEW

**Comment by stwunsch** [comment]
The PR actually fixes two issues at once:

https://sft.its.cern.ch/jira/browse/ROOT-10693
https://sft.its.cern.ch/jira/browse/ROOT-10680

**Comment by stwunsch** [comment]
This is blocked by the issue here: https://sft.its.cern.ch/jira/browse/ROOT-10688

**Comment by stwunsch** [comment]
Merged upstream!

**Comment by stwunsch** [comment]
The issue is resolved by a slightly different impl, see the discussion here: https://bitbucket.org/wlav/cpycppyy/pull-requests/33/return-self-for-__iter__-of-an-iterator/diff#comment-146045830

File: bindings/pyroot_experimental/pythonizations/test/rvec.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        Test the iteration over the iterator of an iterator
+
+        This breaks if __iter__ or tp_iter is not defined for the iterator and causes
+        issues, e.g., in comparison to numpy arrays.
+        '''
+
+        for dtype in self.fundamental_types + self.extra_types:
+            rvec = ROOT.VecOps.RVec[dtype](3)
+            it = iter(rvec)
+            it = iter(it)
```
Comment (etejedor) [inline bindings/pyroot_experimental/pythonizations/test/rvec.py:22]: I think you wanted to save it in another variable `it2 = iter(it)` and then compare both in the assert?

Comment (stwunsch) [reply bindings/pyroot_experimental/pythonizations/test/rvec.py:22]: Oh, thanks :D That's why we do reviews :see_no_evil:

**Comment by stwunsch** [comment]
@etejedor Good to go?

**Pull request merged**